### PR TITLE
Improve security of the kubelet-summary-exporter

### PR DESCRIPTION
**Pull request by Moser-ss**
Use non-root user in the container image

Upgrade dependencies to fix security issues found using Snyk

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,9 +3,9 @@ FROM golang:1.18 as build
 WORKDIR /go/src/app
 COPY . .
 
-RUN go mod download
-RUN CGO_ENABLED=0 make install
-
-FROM gcr.io/distroless/static-debian11
+RUN go mod download \
+  && CGO_ENABLED=0 make install
```
Comment (karlhungus) [inline Dockerfile:7]: These were done on seperate layers on purpose, mind seperating them back out?

Comment (Moser-ss) [reply Dockerfile:7]: Are the layers separated because of caching?

**Comment by Moser-ss** [comment]
Close the PR so the SLA-bot can see the signature

**Review by karlhungus — APPROVED** [review]
Thanks!

File: Dockerfile (hunk 1 above)
Comment (karlhungus) [reply Dockerfile]: Yeah exactly - sorry I should have mentioned, the notion is that we maybe don't often always update libs and code, so we might get some speed out of having modules cached seperarly.

Comment (Moser-ss) [reply Dockerfile]: No problem, also I check your GitHub Actions workflow that publish the container image and the actions build and push is not using cache. Do you mind if I open a PR to add that?

Comment (karlhungus) [reply Dockerfile]: >Do you mind if I open a PR to add that?

Not at all!

**Pull request merged**